src/sqlfluff/core/dialects/dialect_snowflake.py

```diff
@@ -386,6 +386,9 @@ class SelectStatementSegment(ansi_SelectClauseSegment):
 
     parse_grammar = Sequence(
         Ref("SelectClauseSegment"),
+        # Dedent for the indent in the select clause.
+        # It's here so that it can come AFTER any whitespace.
+        Dedent,
```

Comment:
> At parse time, the parser should warn if the indent/dedent level is non-zero at the end of a successful parse.

This is exactly what the "indent balance" warning that has been mentioned a few times is. It's unfortunately happening quite a lot at the moment and templating can throw it off balance quite easily.

> Similar to the super() feature with classes, could we provide support for dialects to extend segments of a parent dialect rather than "replace wholesale"?

I would *love* to do this. I haven't worked out a good way of doing it though. This is definitely worth having an issue for. We do stuff like this quite a lot.

> At dialect module load time, warn if a dialect replaces/overrides a segment from a parent dialect but doesn't have the same "meta" segments.

This is neat idea, I'm not sure how practical it would be to detect though 🤔

Reply:
About the "super()" idea, could it work similarly to "LintFix", where you specify what to delete or edit, or what to add (and where)? The"where" could be specified using "Ref()", e.g. create after "Ref("SelectSegment")".